internal/core/docker_config.go

```diff
@@ -0,0 +1,31 @@
+package core
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/cpuguy83/dockercfg"
+)
+
+// ReadDockerConfig returns the docker config file. It will internally check, in this particular order:
+// 1. the DOCKER_AUTH_CONFIG environment variable, unmarshalling it into a dockercfg.Config
+// 2. the DOCKER_CONFIG environment variable, as the path to the config file
+// 3. else it will load the default config file, which is ~/.docker/config.json
+func ReadDockerConfig() (*dockercfg.Config, error) {
```

Comment:
nit: can we just use `DockerConfig` I try to avoid `Get`, `Read` etc when its not needed to keep things clean.
```suggestion
// DockerConfig returns the docker config file. It will internally check, in this particular order:
// 1. the DOCKER_AUTH_CONFIG environment variable, unmarshalling it into a dockercfg.Config
// 2. the DOCKER_CONFIG environment variable, as the path to the config file
// 3. else it will load the default config file, which is ~/.docker/config.json
func DockerConfig() (*dockercfg.Config, error) {
```